emr-user-role-mapper-application/src/test/com/amazon/aws/emr/integration/IntegrationTestsUserService.java

```diff
@@ -0,0 +1,21 @@
+package com.amazon.aws.emr.integration;
+
+import com.amazon.aws.emr.common.system.impl.CommandBasedPrincipalResolver;
+import com.amazon.aws.emr.common.system.user.UserIdService;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.OptionalInt;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class IntegrationTestsUserService implements UserIdService {
+
+  @Override
+  public OptionalInt resolveSystemUID(String localAddr, int localPort, String remoteAddr,
+      int remotePort, boolean isNativeIMDSApi) {
+     Optional<String> uid = new CommandBasedPrincipalResolver()
+        .runCommand(Arrays.asList("id", "-u")).stream().findFirst();
```

Comment:
Instead of making runCommand public, can we just create a test class that does same thing?

Reply:
I did not want to duplicate the same code. In actuality the `CommandBasedPrincipalResolver` could be static methods, and there is no encapsulation change by changing to public.

Reply:
As we talked to avoid code duplication the access was changed. The method by itself it like a static method with no encapsulation issues.